CAN_Decoder_SourceCode_TestRel_21Dec21.py

```diff
@@ -0,0 +1,119 @@
+import tkinter as tk
+from tkinter import *
+from tkinter import filedialog
+from datetime import datetime
+import re
+
+
+def CAN_Decoder():
+    dataFile = filedialog.askopenfilename()
+    inputFrameData = []
+    
+    # Extract frame from text file using regex approach
+    typicalFrameForm = r"can(\d)\s+([(\w)]{1,3})\s+\[(\d+)\]\s+([\w]{2}(?:\s[\w]{2})*)"
+    frameSize = 0
+    for dataFrame in re.findall(typicalFrameForm, open(dataFile).read()):
+        diagnosticID = int(dataFrame[1], 16)
+        data = list(map(lambda x: int(x, 16), dataFrame[3].split(" ")))
+        frameType = data[0]
+        if frameType <= 6:
+            frameSize = frameType  # Single frame
+        elif frameType == 16:  # 10 for hex notation
+            frameSize = data[1]
+        frameMap = {"DiagnosticID": diagnosticID,
+                    "FrameType": frameType,
+                    "FrameSize": frameSize,
+                    "PayloadData": data}
+        inputFrameData.append(frameMap)
+    extractedData = CAN_ReAssembly(inputFrameData)
+
+    # Output file
+    dateTime = datetime.today().strftime('_%Y%m%d_%H%M%S')
+    outputFileName = dataFile.split(".", 1)[0] + dateTime + '_Output.txt'
+    decodedTextFile = open(outputFileName, "w")
+    for element in extractedData:
+        for item in element:
+            decodedTextFile.write(str(item) + " ")
+        decodedTextFile.write("\n")
+    decodedTextFile.close()
+    print('Decoded file generated. Please check at input file location.')
+
+
+def CAN_ReAssembly(formattedRawData):
+    extractedData = []
+    lineCount = len(formattedRawData)
+    inputData_dict = {}
+    newLineCount = 0
+    cFrameCount = 0
+    for dataLineIndex in range(0, lineCount):
+        dataLine = formattedRawData[dataLineIndex]
+        frameType = dataLine["FrameType"]
+        diagnosticId = dataLine["DiagnosticID"]
+        frameSize = dataLine["FrameSize"]
+        if frameType <= 6:      # Single frame
+            sFrameIndex = 1     # Single frame index
+            payloadData = []
+            for i in range(sFrameIndex, sFrameIndex + frameSize):
```

Comment:
list slicing can come in handy here: `dataLine["PayloadData"][sFrameIndex : sFrameIndex + frameSize + 1]` will extract from `PayloadData` the portion you need. 
https://www.geeksforgeeks.org/python-list-slicing/